sympy/series/formal.py

```diff
@@ -1143,6 +1144,87 @@ def integrate(self, x=None, **kwargs):
 
         return self.func(f, self.x, self.x0, self.dir, (ak, self.xk, ind))
 
+    def convolve(self, other, x=None, n=6, cycle=0, dyadic=False, subset=False):
+        """ Convolute two Formal Power Series and return the truncated terms upto specified order.
+
+        Parameters
+        ==========
+
+        n : Number, optional
+            Specifies the order of the term upto which the polynomial should
+            be truncated.
+        cycle : Integer
+            Specifies the length for doing cyclic convolution.
+        dyadic : bool
+            Identifies the convolution type as dyadic (*bitwise-XOR*)
+            convolution, which is performed using **FWHT**.
+        subset : bool
+            Identifies the convolution type as subset convolution.
+
+        Examples
+        ========
+
+        >>> from sympy import fps, sin, exp, convolution
+        >>> from sympy.abc import x
+        >>> f1 = fps(sin(x))
+        >>> f2 = fps(exp(x))
+
+        >>> f1.convolve(f2, x, 4)
+        x + x**2 + x**3/3 + O(x**4)
+
+        >>> f1.convolve(f2, x, 4, cycle=4)
+        11*x/12 + 35*x**2/36 + x**3/3 + O(x**4)
+
+        >>> f1.convolve(f2, x, 6, dyadic=True)
+        4667/4800 + 2641*x/2880 + x**3/3 + x**4/60 + x**5/20 + O(x**6)
+
+        >>> f1.convolve(f2, x, 6, subset=True)
+        x + x**3/3 + x**5/20 + O(x**6)
+
+        See Also
+        ========
+
+        sympy.discrete.convolutions
+        """
+        if x is None:
+            x = self.x
+        if n is None:
+            return iter(self)
+
+        other = sympify(other)
+
+        if  not isinstance(other, FormalPowerSeries):
+            raise ValueError("Both series should be an instance of FormalPowerSeries"
+                             " class.")
+
+        if self.dir != other.dir:
+            raise ValueError("Both series should be calculated from the"
+                             " same direction.")
+        elif self.x0 != other.x0:
+            raise ValueError("Both series should be calculated about the"
+                             " same point.")
+
+        elif self.x != other.x:
+            raise ValueError("Both series should have the same symbol.")
+
+        if cycle > n:
+            raise ValueError("Value of cycle should be less than or equal to n.")
```

Comment:
This is already checked by ``convolution`` function.

Reply:
Okay. Any other major changes?

Reply:
Except for the convolve API change request, changes look good. It should be ready to merge after the changes. Make sure to update docs and tests according to changes.

Reply:
I believe `convolution` function checks whether the `cycle` is negative or not. If I remove this check, it executes and gives us a wrong answer. So, I think it's better to keep this check here.